crates/sui-graphql-rpc/src/types/object.rs

```diff
@@ -548,6 +587,170 @@ impl Object {
 
         Ok(conn)
     }
+
+    async fn query_live(db: &Db, address: SuiAddress) -> Result<Option<Self>, Error> {
+        use objects::dsl as objects;
+        let vec_address = address.into_vec();
+
+        let stored_obj: Option<StoredObject> = db
+            .execute(move |conn| {
+                conn.first(move || {
+                    objects::objects.filter(objects::object_id.eq(vec_address.clone()))
+                })
+                .optional()
+            })
+            .await
+            .map_err(|e| Error::Internal(format!("Failed to fetch object: {e}")))?;
+
+        stored_obj.map(Self::try_from).transpose()
+    }
+
+    async fn query_at_version(
+        db: &Db,
+        address: SuiAddress,
+        version: u64,
+    ) -> Result<Option<Self>, Error> {
+        use checkpoints::dsl as checkpoints;
+        use objects_history::dsl as history;
+        use objects_snapshot::dsl as snapshot;
+
+        let version = version as i64;
+
+        let results: Option<Vec<StoredHistoryObject>> = db
```

Comment:
There are some scenarios where we would expect a None result, namely when a user provides an object_id for an object that doesn't exist. Otherwise as long as the object_id is valid, it should show up in at least one of snapshot or history